apps/src/templates/studioHomepages/Sections.jsx

```diff
@@ -58,23 +75,33 @@ const Sections = React.createClass({
             result={this.state.sectionsResult}
             nameOrId={this.state.sectionsResultName}
           />
-          {sections.length > 0 && (
+          {isTeacher && numTeacherSections > 0 && sectionFlow2017 && (
+            <SectionsPage
+              className="sectionPage"
+              teacherHomepage={this.props.teacherHomepage}
+            />
+          )}
+          {(
+            (isTeacher && numTeacherSections > 0 && !sectionFlow2017)
+            ||
+            (isStudent && numStudentSections > 0)
+          ) && (
```

Comment:
For this complex of a conditional, I wonder if we're better off precomputing it rather than having it inside our JSX